File: src/main/java/com/udalmik/algs/interviewbit/list/RemoveDuplicatesFromSorted.java
package com.udalmik.algs.interviewbit.list;

public class RemoveDuplicatesFromSorted {

    public ListNode deleteDuplicates(ListNode head) {

        ListNode current = head;
        ListNode prev = null;

        while (current != null) {
            if (prev != null && current.val == prev.val) {
                prev.next = current.next;
            } else {
                prev = current;
            }
            current = current.next;
        }

        return head;
    }

}

File: js/leetcode/linked/remove-duplicates-from-sorted.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var deleteDuplicates = function(head) {

    if (!head || !head.next) return head;
    
    let slow = head;
    let fast = head;

    while (fast) {
        while (fast && fast.val === slow.val) {
            fast = fast.next;
        }
        if (fast) {
            slow.next = fast;
            slow = slow.next;
            fast = fast.next;
        }
    }

    if (slow) {
        slow.next = null;
    }

    return head;

};